src/hooks/screen/useChannelHook.ts

```diff
@@ -3,41 +3,45 @@ import {PostNotificationChannelList} from '../../../types/database/schema/PostNo
 
 function useChannelHook() {
   const determinePostType = (data: PostNotificationChannelList): BaseChannelItemTypeProps => {
-    const isOwningPost = data?.rawJson?.isOwnPost;
+    const isOwnSignedPost =
+      data?.rawJson?.isOwnSignedPost && data?.channelType === 'POST_NOTIFICATION';
+    const isOwnAnonymousPost =
+      data?.rawJson?.isOwnAnonymousPost && data?.channelType === 'ANON_POST_NOTIFICATION';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The conditions `data?.rawJson?.isOwnSignedPost && data?.channelType === 'POST_NOTIFICATION'` and `data?.rawJson?.isOwnAnonymousPost && data?.channelType === 'ANON_POST_NOTIFICATION'` are used to determine if the post is owned by the user and whether it's a signed or anonymous post. However, this could be simplified by introducing a helper function that encapsulates this logic, improving readability and maintainability.

```diff
-    const isOwnSignedPost =
-      data?.rawJson?.isOwnSignedPost && data?.channelType === 'POST_NOTIFICATION';
-    const isOwnAnonymousPost =
-      data?.rawJson?.isOwnAnonymousPost && data?.channelType === 'ANON_POST_NOTIFICATION';

+    const isOwnPostOfType = (postType: string, channelType: string) =>
+      data?.rawJson?.[postType] && data?.channelType === channelType;
+    const isOwnSignedPost = isOwnPostOfType('isOwnSignedPost', 'POST_NOTIFICATION');
+    const isOwnAnonymousPost = isOwnPostOfType('isOwnAnonymousPost', 'ANON_POST_NOTIFICATION');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->